pkg/controller/framework/controller.go

```diff
@@ -105,7 +105,10 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 
 // Returns true once this controller has completed an initial resource listing
 func (c *Controller) HasSynced() bool {
-	return c.config.Queue.HasSynced()
+	if c.config.Queue != nil {
```

Comment:
I agree. A controller shouldn't have nil Queue. There are many places that use Queue directly and checking here not there would make it more fragmented.